client/webserver/http.go

```diff
@@ -145,6 +148,81 @@ func (s *WebServer) handleWallets(w http.ResponseWriter, r *http.Request) {
 	s.sendTemplate(w, "wallets", data)
 }
 
+// handleWalletLogFile is the handler for the '/wallets/logfile' page request.
+func (s *WebServer) handleWalletLogFile(w http.ResponseWriter, r *http.Request) {
+	wf, ok := w.(http.Flusher)
+	if !ok {
+		log.Errorf("unable to flush streamed data")
+		http.Error(w, "unable to flush streamed data", http.StatusBadRequest)
+		return
+	}
+
+	err := r.ParseForm()
+	if err != nil {
+		log.Errorf("error parsing form for wallet log file: %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	assetIDQueryString := r.Form["assetid"]
+	if len(assetIDQueryString) != 1 || len(assetIDQueryString[0]) == 0 {
+		log.Error("could not find asset id in query string")
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	assetID, err := strconv.ParseUint(assetIDQueryString[0], 10, 32)
+	if err != nil {
+		log.Errorf("failed to parse asset id query string %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	logFilePath, err := s.core.GetWalletLogFilePath(uint32(assetID))
+	if err != nil {
+		log.Errorf("failed to get log file path %v", err)
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	logFile, err := os.Open(logFilePath)
+	if err != nil {
+		log.Errorf("error opening log file: %v", err)
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	assetName := dex.BipIDSymbol(uint32(assetID))
+	logFileName := fmt.Sprintf("dcrdex-%s-wallet.log", assetName)
+	w.Header().Set("Content-Disposition", "attachment; filename="+logFileName)
+	w.Header().Set("Content-Type", "text/plain")
+	w.WriteHeader(http.StatusOK)
+
+	bufferLength := int64(1024)
+	buffer := make([]byte, bufferLength)
+	offset := int64(0)
+	for {
+		n, readErr := logFile.ReadAt(buffer, offset)
```

Comment:
I'd typically go with a simple [`io.Copy`](https://pkg.go.dev/io#Copy) here since it essentially abstracts this buffered for loop pattern, with a default 32 kiB buffer.  Was there a particular reason for this approach?

Reply:
No particular reason, I'll switch it

Reply:
Actually I think it makes sense to keep it, because I'm not just copying the file to the `http.ResponseWriter`, but I'm also calling `flush` after each section of the file.

Reply:
If you want.  Can't you just flush at the end though?  It's just that I never bother to do it the long way, never saw a reason.

Reply:
That would happen if we did ReadFile, but io.Copy is buffered.  After one section, it's cannot go on to the next until it's written to the ResponseWriter, and that itself has as a finite write buffer.  If I'm wrong I'd love to learn, but I don't see the file being loaded into memory with io.Copy without explicit flushes after each chunk.

Reply:
Yeah I think you're right. But in that case what's the point of explicitly flushing when generating the csv file?  Wouldn't the `ResponseWriter` buffer fill up there as well, and then get flushed?

Reply:
So this was fun!  Out of curiosity I setup a little file server in a unit test and stepped into the code.  It got to this point in `buffio.(*Writer).Write`

![image](https://user-images.githubusercontent.com/9373513/152008102-021aae50-031c-4660-841f-3f1b753310f8.png)

![image](https://user-images.githubusercontent.com/9373513/152008196-fed58910-9688-4581-bf92-e3c9bebf50d0.png)

That's from `io.Copy` > `net/http.(*response).ReadFrom` > `io.CopyBuffer` > ... > `net/http.(*response).write`.  The `b *Writer` there is a `net/http.chunkWriter` wrapping the actual `net/http.conn` that is alternately `Flush`ed and `Write`n to depending on the state of the buffer.

This was a fun check as I've never really confirmed what paths in the net/http code are actually used.

```go
func TestFile(t *testing.T) {
	f, err := os.Open("/home/jon/dexc-0.4.tar.gz")
	if err != nil {
		t.Fatal(err)
	}
	defer f.Close()

	var server *http.Server

	router := chi.NewRouter()
	router.Get("/", func(w http.ResponseWriter, r *http.Request) {
		io.Copy(w, f)
	})
	router.Get("/shutdown", func(w http.ResponseWriter, r *http.Request) {
		time.AfterFunc(time.Second, func() {
			server.Shutdown(context.Background())
		})
	})

	server = &http.Server{Addr: "127.0.0.1:8888", Handler: router}
	server.ListenAndServe()
}
```

Reply:
Nice, I should do this more often. I'll just remove it from both places then.